network/endpoint.go

```diff
@@ -386,3 +387,25 @@ func (epInfo *EndpointInfo) IsEndpointStateIncomplete() bool {
 	}
 	return false
 }
+
+func (ep *endpoint) validateEndpoint() error {
+	if ep.ContainerID == "" || ep.NICType == "" {
+		return errors.New("endpoint struct must contain a container id and nic type")
```

Comment:
should be :endpoint struct must contain a container id **or** nic type

Reply:
I believe I'm checking that container id is _empty_ or the nic type is _empty_, and if either of those two fields are empty, the error message triggers. If there is just a container id, and nic type is empty, we will satisfy the second condition and error, so both fields must be set (that is, not the empty string) in the endpoint struct to avoid to error.

Reply:
the error message is "and"